From cb136931d824d85c29578e1b112c2bff85d71c77 Mon Sep 17 00:00:00 2001
From: dpeter99 <dpeter99@gmail.com>
Date: Wed, 28 Oct 2020 20:07:25 +0100
Subject: [PATCH] Added task for outputting the forge version

---
 build.gradle | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 71534e2a7..ef58ce6cd 100644
--- a/build.gradle
+++ b/build.gradle
@@ -472,7 +472,29 @@ project(':forge') {
     def changelog = rootProject.file('build/changelog.txt')
     if (changelog.exists())
         extraTxts += changelog
-    
+
+    task printVersion {
+
+
+        // any code that goes here is part of configuring the task
+        // this code will always get run, even if the task is not executed
+        doLast { // add a task action
+
+            def json = [
+                version: project.version
+            ]
+
+            def thingy = new JsonBuilder(json).toPrettyString()
+
+            def output = new File('forge_version.json').write(thingy)
+            //output.write(thingy)
+
+            // any code that goes here is part of executing the task
+            // this code will only get run if and when the task gets executed
+            println thingy
+        }
+    }
+
 	task crowdin(type: CrowdinTask) {
 		key = project.hasProperty('crowdinKey') ? project.crowdinKey : null
 		//TODO: Merge our english.json?
-- 
2.29.1.windows.1

